src/modules/service-worker-remover/index.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  VERSIONED_CACHE_NAME,
+  UNVERSIONED_CACHE_NAME,
+} from '../amp-caching/constants';
+import { cacheName as AMP_ASSET_CACHE } from '../asset-caching/constants';
+import { cacheName as AMP_PUBLISHER_CACHE } from '../document-caching/constants';
+import { cacheName as AMP_PREFETCHED_LINKS } from '../link-prefetch/constants';
+
+export class ServiceWorkerRemover {
+  async installNoOpServiceWorker() {
+    // Taking over the document
+    self.addEventListener('install', function(e: ExtendableEvent) {
+      const { skipWaiting } = self as ServiceWorkerGlobalScope;
+      e.waitUntil(skipWaiting());
+    });
+
+    self.addEventListener('activate', async (e: ExtendableEvent) => {
+      const { clients } = self as ServiceWorkerGlobalScope;
+      e.waitUntil(
+        Promise.all([
+          this.cleanCacheStorage(),
+          this.forceRefreshClients(clients),
+        ]),
+      );
+    });
+  }
+
+  async cleanCacheStorage() {
+    return Promise.all([
```

Comment:
Can we build a test that verifies that an exception thrown in these methods doesn't break the SW?

Reply:
This is a bit tough to do in the current e2e test setup due to the following reasons

- The CacheStorage API never rejects the promise it just returns true/false
- The service worker clears the cache and refreshes the window so even if I try to change the prototype to reject the promise, the service worker thread never gets the change.

Instead I can test this with a unit test case instead.

I'll setup a unit test setup and add a unit tests around `service worker forced nullification` in a separate PR.
It also makes sense to have these tests for `forced nullification` module because this piece is beyond just configuring `workbox`